src/bp/nlu/stan/api.ts

```diff
@@ -131,71 +215,133 @@ export default async function(options: APIOptions, engine: NLUEngine.Engine) {
         }
 
         session = {
-          key: stringId,
           status: 'done',
-          progress: 1,
-          language: model!.id.languageCode
+          progress: 1
         }
       }
 
-      res.send({ success: true, session })
+      const resp: TrainProgressResponseBody = { success: true, session }
+      res.send(resp)
     } catch (err) {
-      res.status(500).send({ success: false, error: err.message })
+      const resp: ErrorResponse = { success: false, error: err.message }
+      res.status(500).send(resp)
     }
   })
 
   router.post('/train/:modelId/cancel', async (req, res) => {
     try {
       const { modelId: stringId } = req.params
-      const { password } = await validateCancelRequestInput(req.body)
+      const { appSecret, appId } = await validateCredentialsFormat(req.body)
```

Comment:
Extract in a middleware?

Reply:
Sure, this could be done, but we kind of know we want to change this part of the API for a better auth system. Let's decide on the header format and then, create a nice auth middleware.